src/components/dialog/dialog.spec.tsx

```diff
@@ -542,6 +543,21 @@ describe("Dialog", () => {
         wrapper.find(StyledDialogTitle)
       );
     });
+
+    it("should allow custom data props on close button to be assigned", () => {
+      const wrapper = enzymeMount(
+        <Dialog
+          open
+          onCancel={() => {}}
+          closeButtonDataProps={{
+            "data-element": "foo",
+            "data-role": "bar",
+          }}
+        />
+      );
+      const closeButton = wrapper.find('[data-component="close"]').at(0);
```

Comment:
comment (non-blocking): I think there'll be only 1 element returned here but could be wrong
```suggestion
      const closeButton = wrapper.find('[data-component="close"]');
```